pkg/collector/corechecks/cluster/ksm/kubernetes_state_aggregators.go

```diff
@@ -147,12 +154,12 @@ func newLastCronJobAggregator() *lastCronJobAggregator {
 func (a *sumValuesAggregator) accumulate(metric ksmstore.DDMetric) {
 	var labelValues [maxNumberOfAllowedLabels]string
 
-	for i, allowedLabel := range a.allowedLabels {
-		if allowedLabel == "" {
+	for i := range a.allowedLabels {
+		if a.allowedLabels[i] == "" {
 			break
 		}
 
-		labelValues[i] = metric.Labels[allowedLabel]
+		labelValues[i] = metric.Labels[a.allowedLabels[i]]
```

Comment:
It's surprising that it's faster to copy the string. I was more worried about the memory usage involved by copying the strings. I do agree that considering this code is not called so often, it might be better to favor readability.